drivers/include/sx127x.h

```diff
@@ -192,21 +192,28 @@ enum {
 };
 
 /**
+ * @name    SX127X device descriptor boolean masks
+ * @{
+ */
+#define SX127X_LOW_DATARATE_OPTIMIZE_FLAG_MASK       (1 << 0)
+#define SX127X_ENABLE_FIXED_HEADER_LENGTH_FLAG_MASK  (1 << 1)
+#define SX127X_ENABLE_CRC_FLAG_MASK                  (1 << 2)
+#define SX127X_CHANNEL_HOPPING_FLAG_MASK             (1 << 3)
+#define SX127X_IQ_INVERTED_FLAG_MASK                 (1 << 4)
+#define SX127X_RX_CONTINUOUS_FLAG_MASK               (1 << 5)
+/** @} */
+
+/**
  * @brief   LoRa configuration structure.
  */
 typedef struct {
+    uint16_t preamble_len;             /**< Length of preamble header */
     uint8_t power;                     /**< Signal power */
     uint8_t bandwidth;                 /**< Signal bandwidth */
     uint8_t datarate;                  /**< Spreading factor rate, e.g datarate */
-    bool low_datarate_optimize;        /**< Optimize for low data rate */
     uint8_t coderate;                  /**< Error coding rate */
-    uint16_t preamble_len;             /**< Length of preamble header */
-    bool use_fix_len;                  /**< Use fixed header len, (enable implicit header mode) */
-    bool crc_on;                       /**< Enable payload CRC */
-    bool freq_hop_on;                  /**< Enable frequency hopping */
     uint8_t freq_hop_period;           /**< Frequency hop period */
-    bool iq_inverted;                  /**< Set inverted IQ */
-    bool rx_continuous;                /**< Use continuous reception */
+    uint8_t flags;                     /**< Boolean flags */
     uint32_t rx_timeout;               /**< RX timeout in symbols */
     uint32_t tx_timeout;               /**< TX timeout in symbols */
```

Comment:
Can you try if moving the two 32-bit numbers to the struct makes a different RAM-wise.